ext/AvizExt/theme.jl

```diff
@@ -7,69 +7,66 @@ const COLORS::Dict{Symbol,Symbol} = Dict(
     :unguided => :lawngreen,
     :guided => :dodgerblue,
+    :target => :blue,
+    :non_target => :red,
     :order => :dodgerblue,
     :topsis => :deepskyblue4,
-    :vikor => :midnightblue
+    :vikor => :midnightblue,
 )
 
-const BINARY_LABELS::Dict{Bool,String} = Dict(0 => "Non-Target", 1 => "Target")
-
-function scenario_type(rs; scenarios=(:))
-    inputs = rs.inputs
-
-    no_seed = (inputs.N_seed_TA .== 0) .& (inputs.N_seed_CA .== 0) .& (inputs.N_seed_SM .== 0)
-    no_fog = inputs.fogging .== 0
-    no_SRM = inputs.SRM .== 0
-    counterfactual = no_seed .& no_fog .& no_SRM
-
-    has_seed = (inputs.N_seed_TA .> 0) .| (inputs.N_seed_CA .> 0) .| (inputs.N_seed_SM .> 0)
-    has_shade = (inputs.fogging .> 0) .| (inputs.SRM .> 0)
-    unguided = (inputs.guided .== 0) .& (has_seed .| has_shade)
-
-    # Guided scenarios must be the inverse of unguided/counterfactual scenarios
-    guided = Bool.(ones(Int64, size(inputs, 1)) .⊻ (counterfactual .| unguided))
-
-    return (counterfactual=counterfactual[scenarios], unguided=unguided[scenarios], guided=guided[scenarios])
+function colors(
+    scen_groups::Dict{<:Any,BitVector}
+)::Dict{Symbol,Union{Symbol,RGBA{Float32}}}
+    group_names = keys(scen_groups)
+    if count(group_names .∉ [keys(COLORS)]) > 0
+        colormap = categorical_colors(:seaborn_bright, length(group_names))
+        return Dict(group => colormap[idx] for (idx, group) in enumerate(group_names))
+    else
+        return Dict(group => COLORS[group] for group in keys(scen_groups))
+    end
 end
-
-function scenario_colors(rs, weight::Float64, hide::BitVector)
-    color_map = fill((COLORS[:guided], weight), size(rs.inputs, 1))
-    scen_type = scenario_type(rs)
-    counterfactual = scen_type.counterfactual
-    unguided = scen_type.unguided
-
-    color_map[counterfactual] .= ((COLORS[:counterfactual], weight),)
-    color_map[unguided] .= ((COLORS[:unguided], weight),)
-
-    if length(hide) > 0
-        color_map[hide] .= ((:white, 0.0),)
+function colors(
+    scen_groups::Dict{Symbol,BitVector}, weight::Float64
+)::Vector{Tuple{Symbol,Float64}}
+    groups = collect(keys(scen_groups))
+    n_scens = length(scen_groups[groups[1]])
+    _colors = Vector{Symbol}(undef, n_scens)
+    scen_colors = colors(scen_groups)
+
+    for (group, scens) in scen_groups
+        _colors[scens] .= [scen_colors[group]]
     end
 
-    return color_map
+    return [(c, weight) for c in _colors]
 end
-function scenario_colors(rs, weight::Float64)
-    color_map = fill((COLORS[:guided], weight), size(rs.inputs, 1))
-
-    scen_type = scenario_type(rs)
-    counterfactual = scen_type.counterfactual
-    unguided = scen_type.unguided
 
-    color_map[counterfactual] .= ((COLORS[:counterfactual], weight),)
-    color_map[unguided] .= ((COLORS[:unguided], weight),)
-
-    return color_map
+function alphas(scen_groups::Dict{Symbol,BitVector})::Dict{Symbol,Float64}
+    return Dict(name => alpha(scens) for (name, scens) in scen_groups)
```

Comment:
I think there's an overuse of dictionaries in this (Dicts are comparably higher in terms of computational cost, as we discussed) but I'm okay with this, at least for now. The effect is probably very low, unless this function is called multiple times in a loop...